core/eth/reader.go

```diff
@@ -916,3 +935,26 @@ func (t *Reader) GetRelayURLs(ctx context.Context) (map[uint32]string, error) {
 
 	return res, nil
 }
+
+func (t *Reader) GetDisperserAddress(ctx context.Context, disperserID uint32) (gethcommon.Address, error) {
+	registry := t.bindings.DisperserRegistry
+	if registry == nil {
+		return gethcommon.Address{}, errors.New("disperser registry not deployed")
+	}
+
+	address, err := registry.DisperserKeyToAddress(
+		&bind.CallOpts{
+			Context: ctx,
+		},
+		disperserID)
```

Comment:
Can we settle on either ID or Key? Using both is unnecessary and adds a little bit confusion.
(if this is meant to be an int, ID seems fit)

Reply:
I'm happy with calling this an "ID". These are serial numbers. Eventually when we allow the community to register dispersers, we will allocate these in monotonic order via a smart contract.

Where do you see the terminology "key" being used in this context?

Reply:
Ah, I think I see what you are getting at. There is a method in the chain reader that references the disperser key:

```
getDisperserKey(
	ctx context.Context,
	now time.Time,
	disperserID uint32) (*gethcommon.Address, error)
```

The disperser key and the disperser ID are actually distinct things.

- Disperser ID: a unique identifier for the disperser, a 32 bit serial number
- Disperser (public) key: the public key used to verify `StoreChunks()` requests from the disperser, is an eth address

The reason why the disperser's public key is not a good identifier for the disperser is that we want the capability to re-key the disperser (e.g. if we lose the key, it gets compromised, or we just want to rotate it).